moveit_core/robot_model/include/moveit/robot_model/link_model.h

```diff
@@ -41,13 +41,16 @@
 #include <vector>
 #include <utility>
 #include <map>
+#include <boost/shared_ptr.hpp>
 #include <Eigen/Geometry>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
 #include <moveit/macros/class_forward.h>
 
 namespace shapes
 {
-MOVEIT_CLASS_FORWARD(Shape);
+class Shape;
+typedef boost::shared_ptr<Shape> ShapePtr;
+typedef boost::shared_ptr<const Shape> ShapeConstPtr;
```

Comment:
I was a bit hesitant to add a macro for declaring `boost::shared_ptr` if they're going to turn in to `std::shared_ptrs` in the not-too-far future. Having macros like that lying around could be seen as an invitation to use them.

How about adding these typedefs in `geometric_shapes` (and possibly a new header `geometric_shapes/forward.h` to avoid including complex header), and until those are available keeping them explicit like this?